presto-main/src/main/java/io/prestosql/operator/scalar/MathFunctions.java

```diff
@@ -607,11 +607,47 @@ public static long random(@SqlType(StandardTypes.BIGINT) long value)
         return ThreadLocalRandom.current().nextLong(value);
     }
 
+    @Description("a pseudo-random number between m and n (exclusive")
+    @ScalarFunction(value = "random", alias = "rand", deterministic = false)
+    @SqlType(StandardTypes.TINYINT)
+    public static long rangeRandomTinyInt(@SqlType(StandardTypes.TINYINT) long initialValue, @SqlType(StandardTypes.TINYINT) long finalValue) {
+
+        checkCondition(initialValue <= finalValue, INVALID_FUNCTION_ARGUMENT, "origin must be less than bound");
+        return ThreadLocalRandom.current().nextInt((int) initialValue, (int) finalValue);
+    }
+
+    @Description("a pseudo-random number between m and n (exclusive")
+    @ScalarFunction(value = "random", alias = "rand", deterministic = false)
+    @SqlType(StandardTypes.SMALLINT)
+    public static long rangeRandomSmallInt(@SqlType(StandardTypes.SMALLINT) long initialValue, @SqlType(StandardTypes.SMALLINT) long finalValue) {
+
+        checkCondition(initialValue <= finalValue, INVALID_FUNCTION_ARGUMENT, "origin must be less than bound");
```

Comment:
- this should be `<`, because from 5 (inclusive) to 5 (exclusive) is an empty range
- wording

```suggestion
        checkCondition(initialValue < finalValue, INVALID_FUNCTION_ARGUMENT, "start value must be less than end value");
```

maybe @electrum will suggest a better wording